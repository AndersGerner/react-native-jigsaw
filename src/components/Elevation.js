import * as React from "react"
import { Animated, StyleSheet } from "react-native"
import shadow from "../styles/shadow"
import { withTheme } from "../core/theming"

class Elevation extends React.Component {
  render() {
    const { style, theme, ...rest } = this.props
    const flattenedStyles = StyleSheet.flatten(style) || {}
    const { elevation } = flattenedStyles

    return <Animated.View {...rest} style={[elevation && shadow(elevation, theme), style]} />
  }
}

export default withTheme(Elevation)
