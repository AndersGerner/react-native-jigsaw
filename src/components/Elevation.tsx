import * as React from "react";
import {
  Animated,
  StyleSheet,
  ViewProps,
  StyleProp,
  ViewStyle,
} from "react-native";
import shadow from "../styles/shadow";
import { withTheme } from "../core/theming";
import themeT from "../styles/DefaultTheme";

interface Props extends ViewProps {
  style?: StyleProp<ViewStyle>;
  theme: typeof themeT;
}

const Elevation: React.FC<Props> = ({ style, theme, ...rest }) => {
  const flattenedStyles = StyleSheet.flatten(style) || {};
  const { elevation } = flattenedStyles;

  const styles = [style];
  if (elevation) {
    styles.push(shadow(elevation, theme));
  }

  return <Animated.View {...rest} style={styles} />;
};

export default withTheme(Elevation);
