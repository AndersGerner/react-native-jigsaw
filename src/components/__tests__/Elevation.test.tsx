import * as React from "react";
import renderer from "react-test-renderer";
import Elevation from "../Elevation";

it("renders elevation", () => {
  const tree = renderer.create(<Elevation />).toJSON();

  expect(tree).toMatchSnapshot();
});
