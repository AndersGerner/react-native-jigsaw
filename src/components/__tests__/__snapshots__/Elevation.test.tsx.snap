// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders elevation 1`] = `
<View
  style={
    Object {
      "zIndex": 100,
    }
  }
/>
`;
